python_modules/dagster/dagster_tests/core_tests/resource_tests/test_resource_definition.py

```diff
@@ -1102,21 +1103,21 @@ def configured_resource(num):
     assert user_code_exc.value.user_exception.args[0] == "beep boop broke"
 
 
-def test_resource_with_enum_in_schema():
-    from enum import Enum as PythonEnum
+class TestPythonEnum(PythonEnum):
+    VALUE_ONE = 0
+    OTHER = 1
 
-    class TestPythonEnum(PythonEnum):
-        VALUE_ONE = 0
-        OTHER = 1
 
-    DagsterEnumType = Enum(
-        "TestEnum",
-        [
-            EnumValue("VALUE_ONE", TestPythonEnum.VALUE_ONE),
-            EnumValue("OTHER", TestPythonEnum.OTHER),
-        ],
-    )
+DagsterEnumType = Enum(
+    "ResourceTestEnum",
+    [
+        EnumValue("VALUE_ONE", TestPythonEnum.VALUE_ONE),
+        EnumValue("OTHER", TestPythonEnum.OTHER),
+    ],
+)
```

Comment:
previously these enums were sharing an object instance but overwriting the inner properties on each `__init__` call

the resulting problem from not overwriting on `__init__` was that the enum `==` checks were failing since the exact enum subclasses are different, being scoped to the different test functions

Reply:
to confirm, that problem is specific to the test setup right?

Reply:
pretty much - any user who has duplicate `Enum` config type definitions that actually point at different in memory python enums would have experienced "last init wins" behavior which is busted just as much as "first init wins" so im not too concerned